src/Plugins/SimplnxCore/src/SimplnxCore/Filters/VerifyTriangleWindingFilter.cpp

```diff
@@ -0,0 +1,192 @@
+#include "simplnx/Filter/Actions/RenameDataAction.hpp"
+#include "simplnx/Parameters/ArraySelectionParameter.hpp"
+#include "simplnx/Parameters/BoolParameter.hpp"
+
+#include "simplnx/Utilities/SIMPLConversion.hpp"
+
+using namespace nx::core;
+
+namespace
+{
+Result<DataPath> FindParentGeometry(const DataPath& path, const DataStructure& dataStructure)
+{
+  auto* maybeTriGeomPtr = dataStructure.getDataAs<TriangleGeom>(path);
+  if(maybeTriGeomPtr != nullptr)
+  {
+    return {path};
+  }
+
+  const DataPath parentPath = path.getParent();
+  if(parentPath.empty())
+  {
+    return {{nonstd::make_unexpected(ErrorCollection{Error{-25740, "Unable to find a Triangle Geometry on the supplied path"}})}};
+  }
+
+  return FindParentGeometry(parentPath, dataStructure);
+}
+
+constexpr StringLiteral k_TempName = "__INTERNAL_!_TEMP_!_PATH__";
+} // namespace
+
+namespace nx::core
+{
+//------------------------------------------------------------------------------
+std::string VerifyTriangleWindingFilter::name() const
+{
+  return FilterTraits<VerifyTriangleWindingFilter>::name.str();
+}
+
+//------------------------------------------------------------------------------
+std::string VerifyTriangleWindingFilter::className() const
+{
+  return FilterTraits<VerifyTriangleWindingFilter>::className;
+}
+
+//------------------------------------------------------------------------------
+Uuid VerifyTriangleWindingFilter::uuid() const
+{
+  return FilterTraits<VerifyTriangleWindingFilter>::uuid;
+}
+
+//------------------------------------------------------------------------------
+std::string VerifyTriangleWindingFilter::humanName() const
+{
+  return "Verify Triangle Winding";
+}
+
+//------------------------------------------------------------------------------
+std::vector<std::string> VerifyTriangleWindingFilter::defaultTags() const
+{
+  return {className(), "Surface Meshing", "Connectivity Arrangement"};
+}
+
+//------------------------------------------------------------------------------
+Parameters VerifyTriangleWindingFilter::parameters() const
+{
+  Parameters params;
+
+  // Create the parameter descriptors that are needed for this filter
+  params.insertSeparator(Parameters::Separator{"Input Face Data"});
+  params.insert(std::make_unique<ArraySelectionParameter>(k_SurfaceMeshFaceLabelsPath_Key, "Face labels Array",
+                                                          "The path to the face labels array, **MUST** reside in target surface mesh (Triangle Geom)", DataPath{},
+                                                          ArraySelectionParameter::AllowedTypes{DataType::int32}, ArraySelectionParameter::AllowedComponentShapes{{2}}));
+
+  params.insertSeparator(Parameters::Separator{"Optional"});
+  params.insertLinkableParameter(
+      std::make_unique<BoolParameter>(k_RepairNormals_Key, "Repair Triangle Normals", "If true we will recalculate normals after execution, disable if no normals exist", true));
```

Comment:
The Description could be: "If enabled, the triangle normals will be recomputed if the triangle normals exist"